tests/example.spec.js

```diff
@@ -16,7 +16,11 @@ test('homepage has Playwright in title and get started link linking to the intro
   // Click the get started link.
   await getStarted.click();
   
-  await page.screenshot({ path: '__assets__/screen_capture.png' });
+  const path = testInfo.outputPath('__assets__/screen_capture.png');
+
+  await page.screenshot({ path });
+
+  testInfo.attachments.push({ name: 'screen_capture.png', path, contentType: 'image/png' });
```

Comment:
Playwright doesn't recommend pushing to the array directly. There's a [method for it](https://playwright.dev/docs/api/class-testinfo#test-info-attach) 💡

Reply:
Moreover, `body` and `path` cannot be set at the same time according to playwright validation `Error: Exactly one of "path" and "body" must be specified`.

If I tried with `body` only
```
const screenshot = await page.screenshot();
await testInfo.attach('screenshot', { body: screenshot, contentType: 'image/png' });
```
Another issue popped up. It complains from our [playwright reporter](https://github.com/saucelabs/sauce-playwright-reporter/blob/main/src/reporter.ts#L323) in this [job](https://app.saucelabs.com/tests/5a771a3d346f4950b91b45ab3c2eb13e)
```
Error in reporter TypeError: The argument 'path' must be a string or Uint8Array without null bytes. Received <Buffer 89 50 4e 47 0d 0a 1a 0a 00 00 00 0d 49 48 44 52 00 00 05 00 00 00 02 d0 08 06 00 00 00 cf 7d dd 56 00 00 20 00 49 44 41 ...
```
And the method that we use is is requiring `path` https://nodejs.org/api/fs.html#fscreatereadstreampath-options which is what we actually cannot provide here.

Reply:
Fixing reporter issue here https://github.com/saucelabs/sauce-playwright-reporter/pull/34/

Reply:
> 

That's because you are creating the screenshot in the `__assets__` folder. Attaching the screenshot to `testInfo` causes it to be copied over to `__assets__` yet again (with a different name), because that's what the output folder points to.

Here's how Playwright puts it:
![Screenshot 2023-04-26 at 3 42 31 PM](https://user-images.githubusercontent.com/1869292/234718598-d43e5911-212e-4806-ab2e-5de7768a954b.png)

Reply:
I see. It works!